test/unit/rx-query.test.ts

```diff
@@ -1582,4 +1584,254 @@ describe('rx-query.test.ts', () => {
+            await cache.setItem(`qc:${queryId}:lwt`, `${now()}`);
+            query1.enablePersistentQueryCache(cache);
+
+            // execute query
+            const result1 = await query1.exec();
+            assert.strictEqual(result1.length, 1);
+
+            const human1Doc = result1[0];
+            await human1Doc.modify(data => {
+              data.age += 1;
+              return data;
+            });
+
+            clearQueryCache(collection);
+
+            const query2 = collection.find({ selector: { age: human1Age }});
+            query2.enablePersistentQueryCache(cache);
+
+            const result2 = await query2.exec();
+
+            assert.strictEqual(result1.length, 1);
+            assert.strictEqual(result2.length, 0);
+            assert.strictEqual(query1._execOverDatabaseCount, 0);
+            assert.strictEqual(query2._execOverDatabaseCount, 0);
+
+            collection.database.destroy();
+        });
+
+        it('does not query from database after adding an object', async () => {
+            const {collection} = await setUpPersistentQueryCacheCollection();
+
+            const human1 = schemaObjects.human();
+            const human2 = schemaObjects.human();
+            const human3 = schemaObjects.human();
+
+            await collection.bulkInsert([human1, human2]);
+
+            const query = collection.find({ limit: 3 });
+            const queryId = query.persistentQueryId();
+            const cache = new Cache();
+            await cache.setItem(`qc:${queryId}`, [human1.passportId, human2.passportId]);
+            await cache.setItem(`qc:${queryId}:lwt`, `${now()}`);
+            query.enablePersistentQueryCache(cache);
+
+            const result1 = await query.exec();
+
+            await collection.insert(human3);
+
+            const result2 = await query.exec();
+
+            assert.strictEqual(result1.length, 2);
+            assert.strictEqual(result2.length, 3);
+            assert.strictEqual(query._execOverDatabaseCount, 0);
+
+            collection.database.destroy();
+        });
+
+        it('does return docs from cache in correct order and with limits applied', async () => {
+            const {collection} = await setUpPersistentQueryCacheCollection();
+
+            const human1 = schemaObjects.human('1', 30);
+            const human2 = schemaObjects.human('2', 40);
+            const human3 = schemaObjects.human('3', 50);
+
+            await collection.bulkInsert([human2, human3]);
+
+            const query1 = collection.find({ limit: 2, sort: [{age: 'asc'}] });
+            const queryId = query1.persistentQueryId();
+            const lwt = now();
+
+            const cache = new Cache();
+            await cache.setItem(`qc:${queryId}`, [human2.passportId, human3.passportId]);
+            await cache.setItem(`qc:${queryId}:lwt`, `${lwt}`);
+
+            await collection.insert(human1);
+
+            clearQueryCache(collection);
```

Comment:
Hm… I think we should use a comment, I am re-using a function that he wrote (https://github.com/readwiseio/rxdb_fork/blob/7607a8b9b2346f5de962eac9d2e04045a36d599c/test/unit/cache-replacement-policy.test.ts#L19). I just extracted it into a helper file, so that I can use it.